documentation/src/main/asciidoc/user_guide/cloud.adoc

```diff
@@ -109,3 +109,101 @@ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -
+
+An example Deployment Configuration (Kubernetes uses very similar concept called `Deployment`) looks like the following:
+
+.DeploymentConfiguration for Rolling Updates
+[source,yaml]
+----
+
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    name: infinispan-cluster
+  spec:
+    replicas: 3
+    strategy:
+      type: Rolling
+      rollingParams:
+        updatePeriodSeconds: 10
+        intervalSeconds: 20
+        timeoutSeconds: 600
+        maxUnavailable: 1
+        maxSurge: 1
+    template:
+      spec:
+        containers:
+        - args:
+          - -Djboss.default.jgroups.stack=kubernetes
+          image: jboss/infinispan-server:latest
+          name: infinispan-server
+          ports:
+          - containerPort: 8181
+            protocol: TCP
+          - containerPort: 8888
+            protocol: TCP
+          - containerPort: 9990
+            protocol: TCP
+          - containerPort: 11211
+            protocol: TCP
+          - containerPort: 11222
+            protocol: TCP
+          - containerPort: 57600
+            protocol: TCP
+          - containerPort: 7600
+            protocol: TCP
+          - containerPort: 8080
+            protocol: TCP
+          env:
+          - name: OPENSHIFT_KUBE_PING_NAMESPACE
+            valueFrom: {fieldRef: {apiVersion: v1, fieldPath: metadata.namespace}}
+          terminationMessagePath: /dev/termination-log
+          terminationGracePeriodSeconds: 90
+          livenessProbe:
+            exec:
+              command:
+              - /usr/local/bin/is_running.sh
+            initialDelaySeconds: 10
+            timeoutSeconds: 80
+            periodSeconds: 60
+            successThreshold: 1
+            failureThreshold: 5
+          readinessProbe:
+             exec:
+                command:
+                - /usr/local/bin/is_healthy.sh
+             initialDelaySeconds: 10
+             timeoutSeconds: 40
+             periodSeconds: 30
+             successThreshold: 2
+             failureThreshold: 5
+
+----
+
+It is also highly recommended to adjust the JGroups stack to discover new nodes (or leaves) more quickly. One should at least
+adjust the value of `FD_ALL` timeout and adjust it to the longest GC Pause.
+
+.Other hints for tuning configuration parameters are:
+* OpenShift should replace running nodes one by one. This can be achieved by adjusting `rollingParams` (`maxUnavailable: 1` and `maxSurge: 1`).
+* Depending on the cluster size, one needs to adjust `updatePeriodSeconds` and `intervalSeconds`. The bigger cluster size is, the bigger those values should be used.
+* When using Initial State Transfer, the `initialDelaySeconds` value for both probes should be set to higher value.
+* During Initial State Transfer nodes might not respond to probes. The best results are achieved with higher values of `failureThreshold` and `successThreshold` values.
+
+==  Rolling upgrades with Kubernetes and OpenShift
```

Comment:
+1, maybe we should use different wording in the titles, like 'Changing configuration'  and 'Data migration' instead of Rolling Updates and Rolling Upgrades